ksql-parser/src/main/java/io/confluent/ksql/parser/properties/with/CreateSourceProperties.java

```diff
@@ -82,10 +84,26 @@ public String getKafkaTopic() {
     return Optional.ofNullable(getString(CreateConfigs.KEY_NAME_PROPERTY));
   }
 
+  @SuppressWarnings("unchecked")
   public Optional<SerdeFactory<Windowed<String>>> getWindowType() {
-    return Optional.ofNullable(getString(CreateConfigs.WINDOW_TYPE_PROPERTY))
-        .map(String::toUpperCase)
-        .map(WINDOW_TYPES::get);
+    final Optional<String> windowType = Optional.ofNullable(
+        getString(CreateConfigs.WINDOW_TYPE_PROPERTY))
+        .map(String::toUpperCase);
+    if (!windowType.isPresent()) {
+      return Optional.empty();
+    }
+    if (SESSION_WINDOW_NAME.equalsIgnoreCase(windowType.get())) {
+      if (getString(CreateConfigs.WINDOW_SIZE_PROPERTY) != null) {
```

Comment:
this check and exception should be done / thrown from the constructor, not the getter.